src/baseclient.cc

```diff
@@ -936,7 +936,8 @@ minio::s3::BaseClient::GetPresignedObjectUrl(GetPresignedObjectUrlArgs args) {
     utils::Time date = utils::Time::Now();
     if (args.request_time) date = args.request_time;
 
-    signer::PresignV4(args.method, url.host, url.path, region, query_params,
+    std::string host = url.HostHeaderValue();
+    signer::PresignV4(args.method, host, url.path, region, query_params,
```

Comment:
```suggestion
    signer::PresignV4(args.method, url.HostHeaderValue(), url.path, region, query_params,
```
To prevent unnecessary string copy

Reply:
It doesn't work as PresignV4() has reference argument.